### PR TITLE
Dart Derping

### PR DESCRIPTION
**Pull request by fundthmcalculus**
This adds support for native dart, need to verify support for flutter (and therefore ios and android) in a separate PR.
ignore build issues with flake or rubocop, those are style checkers.

### CODE REVIEW

File: devops/generate_proto_files.py [hunk 1]
```diff
@@ -122,11 +134,12 @@ def update_python():
 
 def main():
     logging.getLogger().setLevel(logging.INFO)
-    update_golang()
-    update_ruby()
-    update_markdown()
-    update_python()
-    update_java()
+    # update_golang()
```
Comment (geel9) [inline devops/generate_proto_files.py:137]: Seems these should be uncommented

File: devops/generate_proto_files.py [hunk 2]
```diff
@@ -38,13 +38,16 @@ def run_protoc(
     proto_files: Union[List[str], str] = None,
     plugin: str = None,
     protoc_executable: str = "protoc",
+    proto_path: str = None
 ) -> None:
-    proto_path_string = f'--proto_path="{get_language_dir("proto")}"'
+    proto_path_string = f'--proto_path="{get_language_dir(proto_path or "proto")}"'
     plugin_string = f"--plugin={plugin}" if plugin else ""
+    # google_proto_path = f'--proto_path="c:\\bin\\google"'
```
Comment (geel9) [inline devops/generate_proto_files.py:45]: Leftover?

File: python/tests/test_oberon.py [hunk 3]
```diff
@@ -13,7 +14,7 @@
 
 class OberonTests(unittest.TestCase):
     def test_oberon_demo(self):
-        key = oberon.create_key(CreateOberonKeyRequest())
+        key = oberon.create_key(CreateOberonKeyRequest(seed=bytes([1, 2, 3])))
```
Comment (geel9) [inline python/tests/test_oberon.py:17]: Should this be in a separate PR?

File: dart/lib/okapi_native.dart [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    } else if (Platform.isMacOS) {
+      libraryName = path.join("macos", "libokapi.dylib");
+    }
+    // TODO - Support Android, and maybe iOS?
+    libraryPath = path.join(libraryPath, libraryName);
+    final nativeLib = DynamicLibrary.open(libraryPath);
+    return nativeLib;
+  }
+
+  static Pointer<Int8> byteDataToPointer(ByteBuffer byteBuffer) {
```
Comment (geel9) [inline dart/lib/okapi_native.dart:77]: Could this be subject to endianness concerns?

**Review by geel9 — APPROVED** [review]

File: python/tests/test_oberon.py (hunk 3 above)
Comment (fundthmcalculus) [reply python/tests/test_oberon.py:17]: Or removed, was diagnosing something in dart.

File: dart/lib/okapi_native.dart (hunk 4 above)
Comment (fundthmcalculus) [reply dart/lib/okapi_native.dart:77]: Shouldn't be, dart handles that.

File: python/tests/test_oberon.py [hunk 5]
```diff
@@ -13,7 +14,7 @@
 
 class OberonTests(unittest.TestCase):
     def test_oberon_demo(self):
-        key = oberon.create_key(CreateOberonKeyRequest())
+        key = oberon.create_key(CreateOberonKeyRequest()))
```
Comment (geel9) [inline python/tests/test_oberon.py:17]:
```suggestion
        key = oberon.create_key(CreateOberonKeyRequest())
```

**Review by geel9 — CHANGES_REQUESTED** [review]

**Review by geel9 — APPROVED** [review]

**Review by tmarkovski — APPROVED** [review]
